packages/backend/src/auth/controller.ts

```diff
@@ -0,0 +1,82 @@
+import type { Request, Response } from 'express'
+import jwt from 'jsonwebtoken'
+import bcrypt from 'bcrypt'
+import { createLogger } from '../logger'
+import { RequestWithToken } from './middleware'
+import { config } from '../config'
+
+const logger = createLogger(__filename)
+
+export async function createAuthController(): Promise<AuthController> {
+  return {
+    logIn,
+    logOut,
+    getUser,
+  }
+}
+
+export interface AuthController {
+  logIn: (req: Request, res: Response) => Promise<void>
+  logOut: (req: Request, res: Response) => Promise<void>
+  getUser: (req: RequestWithToken, res: Response) => Promise<void>
+}
+
+async function logIn(req: Request, res: Response) {
+  const payload = req.body
+  const { email, password } = payload
+  logger.info(`Login:`, email)
+  if (email === 'admin') {
+    const passwordMatch = await verifyPassword(
+      password,
+      config.auth.adminPasswordHash,
+    )
+    if (passwordMatch) {
+      const token = createToken('admin')
+      res.cookie('token', token, {
+        httpOnly: true,
+        secure: false,
+        sameSite: 'lax',
+      })
+      res.status(200).json({ token })
```

Comment:
```suggestion
      res
          .status(200)
          .cookie('token', token, {
              httpOnly: true,
              secure: false,
              sameSite: 'lax',
          })
          .json({ token })
```

Reply:
Updated. Why do you think it's better in this way? Do you have any reasoning behind? I'm fine with that, just out of curiosity :)

Reply:
No reason necessarily, I just thought that `res`/responses should be final - i.e. a single response per request.
Maybe others would find it confusing as it appears we would respond twice instead of once.

Reply:
Ah, I see. But the response it only one. It's an object you can mutate, similarily to `req`. The response is send only when you cal `send` or `json`. But ok, let's do it in a way you suggest 👍